app/quota.go

```diff
@@ -5,114 +5,109 @@
 package app
 
 import (
-	"github.com/globalsign/mgo"
-	"github.com/globalsign/mgo/bson"
-	"github.com/pkg/errors"
-	"github.com/tsuru/tsuru/db"
-	"github.com/tsuru/tsuru/quota"
+	appTypes "github.com/tsuru/tsuru/types/app"
 )
 
-func reserveUnits(app *App, quantity int) error {
-	app, err := checkAppLimit(app.Name, quantity)
+type appQuotaService struct {
+	storage appTypes.QuotaStorage
+}
+
+// ReserveUnits implements ReserveUnits method from AppQuotaService interface
+func (s *appQuotaService) ReserveUnits(appName string, quantity int) error {
+	quota, err := s.storage.FindByAppName(appName)
 	if err != nil {
+		if err == appTypes.ErrAppNotFound {
```

Comment:
I agree that should just keep one. Just thought it could be out of the scope of this PR. Since it's just changing some references without any major logic change, I'm moving forward with changing to only the one in `appTypes`.